site/app/models/RainbowCustomization.php

```diff
@@ -423,17 +423,32 @@ public function getSectionsAndLabels() {
             // Get sections from the file
             $sectionsFromFile = (array) $this->RCJSON->getSection();
 
-            // If sections from database is larger than sections from file then there must be a new section in
-            // in the database, add new fields into sections from file with defaults
-            $sectionsFromFileCount = count($sectionsFromFile);
+            // gets the number of sections in the database
             $sectionsCount = count($sections);
 
-            if ($sectionsFromFileCount != $sectionsCount) {
-                for ($i = $sectionsFromFileCount + 1; $i <= $sectionsCount; $i++) {
-                    $sectionsFromFile[$i] = (string) $i;
+            // creates a copy of the database sections to compare against the file sections
+            $temp = [];
+            for ($i = 0; $i < $sectionsCount; $i++) {
+                $temp[$db_sections[$i]['sections_registration_id']] = (string) $db_sections[$i]['sections_registration_id'];
+            }
+
+            // checks whether or not a sections has been added and upodates the file sections
+            $result = array_diff_key($temp, $sectionsFromFile);
+            if ($result) {
```

Comment:
We know the return type of array_diff_key to be an array so the if statement would check if it's empty or not. For someone looking at this code for the first time, it would probably be better to do something like this instead:
```suggestion
            if (count($result) > 0) {
```

Do the same for line 445.